io/src/main/scala/sbt/internal/io/LazyFileAttributes.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * sbt IO
+ *
+ * Copyright 2011 - 2019, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ */
+
+package sbt.internal.io
+import java.nio.file.attribute.{ BasicFileAttributes, FileTime }
+import java.nio.file.{ Files, Path }
+
+import scala.util.Try
+
+private[sbt] class LazyFileAttributes(path: Path,
```

Comment:
Could you add Scaladoc documenting the purpose and state of this class?
I am not sure if lazy is a good name for it, if it caches the result forever. Maybe it should be called `FileAttributesOnce`?

Reply:
Good point. I think I might back out of extending BasicFileAttributes for now and only implement SimpleFileAttributes.

Reply:
I was still thinking about implementing the nio FileSystem when I wrote this but I think that is overkill for 1.3.0.